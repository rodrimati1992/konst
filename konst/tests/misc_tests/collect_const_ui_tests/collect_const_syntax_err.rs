const _: () = {
    konst::iter::collect_const!{() => 0..10,all(|_|false)};
};

const _: () = {
    konst::iter::collect_const!{() =>};
};

const _: () = {
    konst::iter::collect_const!{};
};

fn main(){}