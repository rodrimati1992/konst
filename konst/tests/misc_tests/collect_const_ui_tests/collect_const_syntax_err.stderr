error: the `all` method cannot be called in this macro
 --> tests/misc_tests/collect_const_ui_tests/collect_const_syntax_err.rs:2:5
  |
2 |     konst::iter::collect_const!{() => 0..10,all(|_|false)};
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::iter::__cim_method_not_found_err` which comes from the expansion of the macro `konst::iter::collect_const` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected iterator argument
 --> tests/misc_tests/collect_const_ui_tests/collect_const_syntax_err.rs:6:5
  |
6 |     konst::iter::collect_const!{() =>};
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::__process_iter_args` which comes from the expansion of the macro `konst::iter::collect_const` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of macro invocation
  --> tests/misc_tests/collect_const_ui_tests/collect_const_syntax_err.rs:10:5
   |
10 |     konst::iter::collect_const!{};
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing tokens in macro arguments
   |
note: while trying to match meta-variable `$Item:ty`
  --> $WORKSPACE/konst_kernel/src/collect_const.rs
   |
   |         $Item:ty =>
   |         ^^^^^^^^

error[E0308]: mismatched types
 --> tests/misc_tests/collect_const_ui_tests/collect_const_syntax_err.rs:2:5
  |
2 |     konst::iter::collect_const!{() => 0..10,all(|_|false)};
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
  |
  = note: this error originates in the macro `$crate::iter::__cim_preprocess_methods` which comes from the expansion of the macro `konst::iter::collect_const` (in Nightly builds, run with -Z macro-backtrace for more info)
